Saas cloud storage config per dev env

drigato opened the pull request:
The cloud_storage.conf file will now be stored under /etc/airtime-saas/<dev_env>
Each development environment can have its own copy of cloud_storage.conf.

Along with this change I also renamed rabbitmq-analyzer-<dev_env>.ini to rabbitmq-analyzer.ini, which will also be stored under /etc/airtime-saas/<dev_env>

drigato committed:
SAAS-560: Deploy separate cloud storage config files for each development environment

drigato committed:
Analyzer log statements for debugging

drigato committed:
More analyzer log statements

drigato committed:
SAAS-560: Deploy separate cloud storage config files for each development environment

Changed analyzer upstart to take the cloud storage config file as a command line option
Dropped the dev env portion from the rabbitmq-analyzer.ini filename

drigato committed:
Merge branch 'saas' into saas-cloud-storage-config-per-dev-env

Conflicts:

python_apps/airtime_analyzer/airtime_analyzer/analyzer_pipeline.py

python_apps/airtime_analyzer/airtime_analyzer/cloud_storage_uploader.py

python_apps/airtime_analyzer/airtime_analyzer/message_listener.py

drigato committed:
SAAS-560: Deploy separate cloud storage config files for each development environment

Small fix

drigato committed:
SAAS-560: Deploy separate cloud storage config files for each development environment

Renamed the analyzer config variables to be named specially for rabbitmq
config

drigato committed:
Merge branch 'saas' into saas-cloud-storage-config-per-dev-env

asantoni commented on airtime_mvc/application/configs/conf.php:
```diff
@@ -54,6 +41,19 @@ public static function loadConfig() {
+        // Tells us where file uploads will be uploaded to.
+        // It will either be set to a cloud storage backend or local file storage.
+        $CC_CONFIG["current_backend"] = $cloudStorageValues["current_backend"]["storage_backend"];
+        
```
Can you make this code do one more thing:
- Default to the "production" cloud_storage.conf if a per-environment conf doesn't exist? (This is a pattern we should always be trying to implement, I think)

If this was in regular Airtime, I'd also say that we should make the code still work and fallback to file storage if no cloud_storage.conf was found at all. However, I think that since this code is saas-only and we ONLY want S3 uploads now, I'd rather have the code fail out (like it does now) if a cloud_storage.conf wasn't found. (If that file doesn't exist at all, then it's a server configuration issue and we'd want to know about that right away, rather than having files silently be stored on disk again.)

drigato replied:
Done.

asantoni commented on airtime_mvc/application/models/RabbitMq.php:
```diff
@@ -89,7 +89,7 @@ public static function SendMessageToAnalyzer($tmpFilePath, $importedStorageDirec
             $devEnv = $CC_CONFIG["dev_env"];
         }
-        $config = parse_ini_file("/etc/airtime-saas/rabbitmq-analyzer-" . $devEnv . ".ini", true);
+        $config = parse_ini_file("/etc/airtime-saas/".$devEnv."/rabbitmq-analyzer.ini", true);
```
Should we make this fallback to the production rabbitmq config if that file doesn't exist here too?

asantoni commented on python_apps/airtime_analyzer/airtime_analyzer/analyzer_pipeline.py:
```diff
@@ -21,7 +22,7 @@ class AnalyzerPipeline:
     
     @staticmethod
-    def run_analysis(queue, audio_file_path, import_directory, original_filename, file_prefix, cloud_storage_enabled):
+    def run_analysis(queue, audio_file_path, import_directory, original_filename, file_prefix, cloud_storage_config):
```
Can you fix the unit tests?

drigato replied:
Done.

drigato committed:
SAAS-560: Deploy separate cloud storage config files for each development environment

Default to production config files if dev env specific files are not
found.
Fix analyzer unit tests.

asantoni commented:
Looks good to me now, thanks!
